### PR TITLE
Two version-related improvements

### PR DESCRIPTION
**Pull request by jll63**
* Add command line switch (`--version` or `-v`) to query broker version.
* Let the broker set `appConfig.brokerVersion` from compiled-in `bmqbrkrscm_version`.

**Testing performed**
* `bmqbrkr.tsk --version` and `bmqbrkr.tsk -v`: displays the version
* `bmqbrkr.tsk`: print previous error message (missing config) and exit - as before
* Execute the `run` script: works as before

### CODE REVIEW

**Comment by pniedzielski** [comment]
@jll63, It looks like your git email isn't set up correctly, so the Sanity Check is failing.

File: src/applications/bmqbrkr/bmqbrkr.m.cpp [hunk 1]
```diff
@@ -299,6 +299,8 @@ static int getConfig(bsl::ostream&      errorDescription,
     }
 
     taskEnv->d_config.appConfig().etcDir() = configDir;
+    taskEnv->d_config.appConfig().brokerVersion() =
+        bmqbrkrscm::Version::versionAsInt();
```
Comment (quarter-note) [inline src/applications/bmqbrkr/bmqbrkr.m.cpp:303]: So, before this, `brokerVersion` field was being populated during config decoding step at line 287 above?

File: src/applications/bmqbrkr/bmqbrkr.m.cpp [hunk 2]
```diff
@@ -592,50 +594,22 @@ run(bsl::ostream& errorDescription, TaskEnvironment* taskEnv, bool wait = true)
...
+    int         port    = 0;
+    bool        version = false;
 
     balcl::OptionInfo specTable[] = {
         {"",
          "config",
          "Path to the configuration directory",
          balcl::TypeInfo(&configDir),
-         balcl::OccurrenceInfo::e_REQUIRED},
+         balcl::OccurrenceInfo::e_OPTIONAL},
```
Comment (quarter-note) [inline src/applications/bmqbrkr/bmqbrkr.m.cpp:612]: Why is this changed to optional?  So that you can invoke `/path/to/bmqbrkr -v` to check its version, w/o the command line parser complaining about the config dir?

File: src/applications/bmqbrkr/bmqbrkr.m.cpp [hunk 3]
```diff
@@ -661,6 +635,11 @@ int main(int argc, const char* argv[])
          "Override port",
          balcl::TypeInfo(&port),
          balcl::OccurrenceInfo::e_OPTIONAL},
+        {"v|version",
+         "version",
+         "Show version number",
```
Comment (quarter-note) [inline src/applications/bmqbrkr/bmqbrkr.m.cpp:640]: May change to "Show broker version number"

File: src/applications/bmqbrkr/bmqbrkr.m.cpp [hunk 4]
```diff
@@ -672,6 +651,46 @@ int main(int argc, const char* argv[])
         return mqbu::ExitCode::e_COMMAND_LINE;  // RETURN
     }
 
+    if (version) {
+        bsl::cout << bmqbrkrscm::Version::version() << "\n";
```
Comment (quarter-note) [inline src/applications/bmqbrkr/bmqbrkr.m.cpp:655]: May be add a string like "BlazingMQ broker version: " before printing the version. Unless, you just want the broker to print the version so that some scripts can rely on the output?

File: src/applications/bmqbrkr/bmqbrkr.m.cpp [hunk 5]
```diff
@@ -672,6 +651,46 @@ int main(int argc, const char* argv[])
...
+        return 0;
+    }
+
+    if (configDir.empty()) {
+        bsl::cerr << "Error: No value supplied for the non-option argument "
+                     "\"config\".\n";
```
Comment (quarter-note) [inline src/applications/bmqbrkr/bmqbrkr.m.cpp:661]: Broker should exit if no cfg dir is specified, right?

File: src/applications/bmqbrkr/bmqbrkr.m.cpp [hunk 6]
```diff
@@ -672,6 +651,46 @@ int main(int argc, const char* argv[])
...
+        bsl::cerr << "Failed to install SIGINT handler  (rc: " << rc << ")\n"
+                  << bsl::flush;
+    }
+
+    // If running in a TTY, enable signal handler so that user signals can be
+    // used to stop the task.
+    if (isatty(fileno(stdout))) {
+        ::sigaction(SIGQUIT, &sa, NULL);
+        ::sigaction(SIGTERM, &sa, NULL);
+    }
```
Comment (quarter-note) [inline src/applications/bmqbrkr/bmqbrkr.m.cpp:689]: Any reason to move the signal handling related logic here instead of keeping it earlier?

**Review by quarter-note — COMMENTED** [review]
Some comments

File: src/applications/bmqbrkr/bmqbrkr.m.cpp (hunk 1 above)
Comment (jll63) [reply src/applications/bmqbrkr/bmqbrkr.m.cpp:303]: Yes.

File: src/applications/bmqbrkr/bmqbrkr.m.cpp (hunk 2 above)
Comment (jll63) [reply src/applications/bmqbrkr/bmqbrkr.m.cpp:612]: Yes. The test for the `config` argument is done "manually" if `-v` is not passed.

File: src/applications/bmqbrkr/bmqbrkr.m.cpp (hunk 4 above)
Comment (jll63) [reply src/applications/bmqbrkr/bmqbrkr.m.cpp:655]: Anyway the more verbose output is not difficult to parse. Adding the string.

File: src/applications/bmqbrkr/bmqbrkr.m.cpp (hunk 5 above)
Comment (jll63) [reply src/applications/bmqbrkr/bmqbrkr.m.cpp:661]: Ah yes.

File: src/applications/bmqbrkr/bmqbrkr.m.cpp (hunk 6 above)
Comment (jll63) [reply src/applications/bmqbrkr/bmqbrkr.m.cpp:689]: We don't need to do that (nor to print "STARTING...")  if we are just printing the version.

**Comment by jll63** [comment]
All comments addressed. Changes in separate commit, to be squashed upon merging.

File: src/applications/bmqbrkr/bmqbrkr.m.cpp (hunk 4 above)
Comment (quarter-note) [reply src/applications/bmqbrkr/bmqbrkr.m.cpp]: If the main goal is to let some script invoke `/path/to/bmqbrkr -v` to figure out the version programmatically, then perhaps we should print only the verison w/o any string.  Will leave it up to you to make that call.

**Review by quarter-note — APPROVED** [review]
Left a comment.  Feel free to merge once CI is happy.
